lib/executor/helpers.go

```diff
@@ -72,6 +72,16 @@ func validateStages(stages []Stage) []error {
 		} else if s.Target.Int64 < 0 {
 			errors = append(errors, fmt.Errorf("the target for stage %d shouldn't be negative", stageNum))
 		}
+
+		switch s.Progression.String {
```

Comment:
Considering `s.Progression` is defined as a `null.String` I would follow the same procedure as for the other field, and at least append an error if `s.Target.Valid` is `false.